FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/viewmodel/ComposeMsgViewModel.kt

```diff
@@ -389,7 +387,7 @@ class ComposeMsgViewModel(isCandidateToEncrypt: Boolean, application: Applicatio
       email: String,
       activeAccount: AccountEntity?
     ): Result<PubResponse> = withContext(lookUpLimitedParallelismDispatcher) {
-      return@withContext apiRepository.pubLookup(
+      return@withContext ApiClientRepository.Attester.pubLookup(
```

Comment:
Excellent, much clearer. But the name of this `private suspend fun pubLookup` method is also ambiguous, should be `private suspend fun attesterPubLookup` on line 386

Also, how come I don't see WKD mentioned in this file?

Reply:
We have [WkdClient](https://github.com/FlowCrypt/flowcrypt-android/blob/c914c1f9a771b52607dc321e77cb67665eda742f/FlowCrypt/src/main/java/com/flowcrypt/email/api/wkd/WkdClient.kt) for that.
ref https://github.com/FlowCrypt/flowcrypt-android/pull/1377

As I remember you had preferred to have all logic in a single class. That's why I didn't move some methods to `FlowCryptApiRepository`

Reply:
Relooked at the repo. I can't find such a comment. Maybe I'm wrong.

Reply:
ref https://github.com/FlowCrypt/flowcrypt-android/pull/1399

Reply:
Single class, yes. It was called `PubLookup` class. But here we are calling only `Attester`. If it was calling `PubLookup` here, I wouldn't say anything, because I know that one uses all appropriate public key sources.

Why is it only calling `Attester` here? Anytime any public key is being looked up externally, it should always be with `PubLookup` (with the exception of Attester view in Settings)

Reply:
> Excellent, much clearer. But the name of this private suspend fun pubLookup method is also ambiguous, should be private suspend fun attesterPubLookup on line 386

I've relooked over the code. All this is wrong. From the beginning. This method should be called `pubLookup` as it was before. It describes what it does - look up a public key from different sources(WKD + attester).

We have the wrong definition in ApiClientRepository.Attester.pubLookup. Because of ApiClientRepository.Attester.pubLookup describes the logic of fetching a pub key from different sources(WKD + attester).

I think it will be clearer to move this code to a separate class. For example, `PubLookup`

Reply:
Now we have `ApiClientRepository.PubLookup.fetchPubKey`

Reply:
Yes, this is the importance of naming every method with the source it works with. Else it's hard to review and ensure.